### PR TITLE
only run `go install/build` or `go test`

### PR DESCRIPTION
**Pull request by lloiser**
On smaller projects, running both may seem like a reasonable approach because both commands finish in less than 1 second. For larger projects, though a simple go build can take several seconds especially if you include C code.

TODOs:

- [x] The only downside I can think of at the moment is autocompletion... I will have to test this for a while. Update: go test supports installing all dependencies via -i
- [x] Maybe kill previous install/test of the same folder if it has not finished yet.

### CODE REVIEW

**Comment by zmb3** [comment]
This one good to go too?

**Comment by lloiser** [comment]
I have tested it for a while now so it seems to work :)

File: lib/build/builder.js [hunk 1]
```diff
@@ -239,14 +233,14 @@ class Builder {
           // we'll take care of this one, skip over the -o flag
           i++
           continue
-        } else if (parsed[i] === '-c') {
+        } else if (parsed[i] === '-c' || parsed[i] === '-i') {
```
Comment (zmb3) [inline lib/build/builder.js]: Which version of Go added the `-i` flag to `go test`?  Is it supported in 1.10?

Comment (lloiser) [reply lib/build/builder.js]: [1.9](https://github.com/golang/go/blob/release-branch.go1.9/src/cmd/go/internal/test/test.go#L104), [1.10](https://github.com/golang/go/blob/release-branch.go1.10/src/cmd/go/internal/test/test.go#L157) and obviously [1.11](https://github.com/golang/go/blob/release-branch.go1.11/src/cmd/go/internal/test/test.go#L154).

Interestingly this even dates back to [1.0](https://github.com/golang/go/blob/release-branch.go1/src/cmd/go/test.go#L62)!

**Pull request merged**